evap/staff/views.py

```diff
@@ -732,14 +735,63 @@ def course_person_import(request, semester_id, course_id):
 def helper_person_import(users, course, import_type):
     if import_type == 'participant':
         course.participants.add(*users)
-        return _("{} Participants added to course {}").format(len(users), course.name)
+        msg = _("{} Participants added to course {}").format(len(users), course.name)
+        for user in users:
+            msg += "<br>"
+            msg += "{} {} ({})".format(user.first_name, user.last_name, user.username)
+        return mark_safe(msg)
+
     else:
         for user in users:
             order = Contribution.objects.filter(course=course).count()
             contribution = Contribution(course=course, contributor=user, order=order)
             contribution.save()
             course.contributions.add(contribution)
-        return _("{} Contributors added to course {}").format(len(users), course.name)
+        msg = _("{} Contributors added to course {}").format(len(users), course.name)
+        for user in users:
+            msg += "<br>"
+            msg += "{} {} ({})".format(user.first_name, user.last_name, user.username)
+        return mark_safe(msg)
+
+
+def helper_create_test_success_message(import_type, course, imported_users):
+    if import_type == "participant":
+        msg = _("{} participants would be added to the course {}:").format(len(imported_users), course.name)
+    else:  # import_type == "contributor"
+        msg = _("{} contributors would be added to the course {}:").format(len(imported_users), course.name)
+
+    for user in imported_users:
+        msg += "<br>"
+        msg += "{} {} ({})".format(user.first_name, user.last_name, user.username)
+
+    return mark_safe(msg)
+
+
+def helper_handle_already_related(import_type, course, user_list, warnings):
+    already_related = []
+
+    if import_type == "participant":
+        for new_participant in user_list:
+            if new_participant in course.participants.all():
+                already_related.append(new_participant)
+    else:  # import_type == "contributor"
+        for new_contributor in user_list:
+            if Contribution.objects.filter(course=course, contributor=new_contributor).exists():
+                already_related.append(new_contributor)
+
+    if already_related:
+        if import_type == "participant":
+            msg = _("The following {} user(s) are already course participants in course {}:").format(len(already_related), course.name)
+        else:  # import_type == "contributor"
+            msg = _("The following {} user(s) are already contributing to course {}:").format(len(already_related), course.name)
+
+        for participant in already_related:
+            msg += "<br>"
+            msg += ("{} {} ({})").format(participant.first_name, participant.last_name, participant.username)
```

Comment:
The importer does not know about any of the participant or contributor stuff. It just handles the user import. All the course related importing is happening in the view then, so this is the innermost place where the warnings could be generated.